### PR TITLE
Try async for fam events

### PR DESCRIPTION
**Pull request by TordJoranger**

### CODE REVIEW

File: src/Equinor.ProCoSys.BusSender.Core/Services/BusSenderService.cs [hunk 1]
```diff
@@ -70,20 +70,93 @@ public async Task HandleBusEvents()
...
     }
 
+    private async Task ProcessBusEvents(List<BusEvent> events)
+    {
+        events = SetDuplicatesToSkipped(events);
+        var dsw = Stopwatch.StartNew();
+
+        var unProcessedEvents = events.Where(busEvent => busEvent.Status == Status.UnProcessed).ToList();
+        _logger.LogInformation("Amount of messages to process: {count} ", unProcessedEvents.Count);
+        // await unProcessedEvents
```
Comment (TordJoranger) [inline src/Equinor.ProCoSys.BusSender.Core/Services/BusSenderService.cs:83]: Shall remove (soon) Maybe I want to make one more attempt at getting parallel DB calls to work

File: src/Equinor.ProCoSys.PcsBus/Sender/PcsBusSender.cs [hunk 2]
```diff
@@ -38,4 +38,18 @@ public async Task CloseAllAsync()
             await topicClient.Value.CloseAsync();
         }
     }
+
+    public async ValueTask<ServiceBusMessageBatch> CreateMessageBatchAsync(string topic)
+    {
+        var sender = _busSenders.SingleOrDefault(t => t.Key == topic).Value;
```
[the remark this answers is not in the document]
Comment (TordJoranger) [reply src/Equinor.ProCoSys.PcsBus/Sender/PcsBusSender.cs:44]: per design ;)

File: src/Equinor.ProCoSys.PcsBus/Sender/PcsBusSender.cs [hunk 3]
```diff
@@ -38,4 +38,18 @@ public async Task CloseAllAsync()
...
+        var serviceBusMessageBatch = await sender.CreateMessageBatchAsync();
+        return serviceBusMessageBatch;
+    }
+
+    public async Task SendMessagesAsync(ServiceBusMessageBatch messageBatch, string topic)
+    {
+        var sender = _busSenders.SingleOrDefault(t => t.Key == topic).Value;
```
[the remark this answers is not in the document]
Comment (TordJoranger) [reply src/Equinor.ProCoSys.PcsBus/Sender/PcsBusSender.cs:51]: Sure about this? Isn't value = null? if no topic found? (as in default for KVP = new KVP(null,null) )

File: src/Equinor.ProCoSys.PcsBus/Queries/PipeTestQuery.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        from pipetest pt
+            join pipingrevision pr on pr.pipingrevision_id = pt.pipingrevision_id      
+            join pipetestlibrary ptl on ptl.pipetestlibrary_id = pt.pipetestlibrary_id  
+        {whereClause}";
+    }
+
+    private static string CreateWhereClause(long? revisionId, long? pipeTestLibraryId, string plant)
+    {
+        var whereClause = "";
+        if (revisionId is not null & pipeTestLibraryId is not null & plant is not null)
```
[the remark this answers is not in the document]
Comment (TordJoranger) [reply src/Equinor.ProCoSys.PcsBus/Queries/PipeTestQuery.cs:28]: I've not been able to chain "and" "or" etc. would && be better, or are you thinking methods?

File: src/Equinor.ProCoSys.PcsBus/Sender/PcsBusSender.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (jsoi-eq) [reply src/Equinor.ProCoSys.PcsBus/Sender/PcsBusSender.cs:55]: You are right, it won't crash since KeyValuePair is a struct.

**Review by jsoi-eq — APPROVED** [review]

**Pull request merged**